File: DroidCafe/app/src/main/java/com/example/droidcafe/MainActivity.java
package com.example.droidcafe;

import android.content.Intent;
import android.os.Bundle;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;

import android.view.View;

import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

import android.widget.Toast;

import com.google.android.material.floatingactionbutton.FloatingActionButton;

public class MainActivity extends AppCompatActivity {
//    private FloatingActionButton fab_cart;
    private String mOrderMessage;

    public static final String EXTRA_MESSAGE = "com.example.android.droidcafe.extra.MESSAGE";

    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.content_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });

//        fab_cart = (FloatingActionButton)findViewById(R.id.fab_shopping_cart);
//        fab_cart.setOnClickListener(new View.OnClickListener() {
//            @Override
//            public void onClick(View v) {
//                Intent intent = new Intent(MainActivity.this, OrderActivity.class);
//                startActivity(intent);
//            }
//        });


        mOrderMessage = getString(R.string.donut_order_message);
        displayToast(mOrderMessage);

    }

    public void displayToast(String message) {
        Toast.makeText(getApplicationContext(), message, Toast.LENGTH_SHORT).show();
    }

    public void showDonut(View view) {
        displayToast(getString(R.string.donut_order_message));
    }

    /**
     * Shows a message that the donut image was clicked.
     */
    public void showDonutOrder(View view) {
        displayToast(getString(R.string.donut_order_message));
    }

    /**
     * Shows a message that the ice cream sandwich image was clicked.
     */
    public void showIceCreamOrder(View view) {
        displayToast(getString(R.string.ice_cream_order_message));
    }

    /**
     * Shows a message that the froyo image was clicked.
     */
    public void showFroyoOrder(View view) {
        displayToast(getString(R.string.froyo_order_message));
    }
    public void onClick(View v) {
        Intent intent = new Intent(MainActivity.this, OrderActivity.class);
        intent.putExtra(EXTRA_MESSAGE, mOrderMessage);
        startActivity(intent);
    }
}

File: CountdownTimerApp/app/src/main/java/com/example/countdowntimerapp/MainActivity.kt
package com.example.countdowntimerapp

import android.os.Bundle
import android.os.Handler
import android.os.Looper
import androidx.appcompat.app.AppCompatActivity
import android.widget.TextView
import com.example.countdowntimerapp.R

class MainActivity : AppCompatActivity() {

    private lateinit var tvTimer: TextView
    private var timeElapsed = 0 // Thời gian đã trôi qua
    private val handler = Handler(Looper.getMainLooper())

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        tvTimer = findViewById(R.id.tvTimer)

        // Bắt đầu thread đếm thời gian
        startTimerThread()
    }

    private fun startTimerThread() {
        Thread {
            while (true) { // Vòng lặp vô hạn để đếm thời gian
                Thread.sleep(1000) // Dừng 1 giây
                timeElapsed++ // Tăng thời gian

                // Cập nhật giao diện qua Handler
                handler.post {
                    tvTimer.text = "$timeElapsed giây"
                }
            }
        }.start()
    }
}
